### DISCUSSION TITLE
using entity ruler "id" ent_id field in trainable NER

### DISCUSSION THREAD

**Discussion by jaiqueguille** [explosion/spaCy#8020] [Help: Coding & Implementations] [answered]
Can the ent_id_ field available to the entity ruler be passed to the NER? My current thinking is to start with a rules-based entity ruler to add to the pipe via a patterns.json file before using the trainable NER. I don't see a mention of the id attribute for NER in the docs.

The id attribute as described for the entity ruler: 

> `ruler = nlp.add_pipe("entity_ruler")
> patterns = [{"label": "ORG", "pattern": "Apple", "id": "apple"},
>             {"label": "GPE", "pattern": [{"LOWER": "san"}, {"LOWER": "francisco"}], "id": "san-francisco"},
>             {"label": "GPE", "pattern": [{"LOWER": "san"}, {"LOWER": "fran"}], "id": "san-francisco"}]
> ruler.add_patterns(patterns)`

Docs with spans added to create the training data for the entity recognizer only reference the label, not the id field:

> `# Training data for an entity recognizer (option 2)
> doc = nlp("Laura flew to Silicon Valley.")
> gold_dict = {"entities": [(0, 5, "PERSON"), (14, 28, "LOC")]}
> example = Example.from_dict(doc, gold_dict)`

Is it possible to also include the "id "attribute in the training data to allow multiple patterns to be associated with the same entity in a trainable model?

e.g.

> `("iPhone 11 vs iPhone 8: What's the difference?", {'entities': [(0, 9, 'GADGET', "id: Phone"), (13, 21, 'GADGET', "id: Phone")]})`

**Comment by polm** [comment, marked as answer]
No, the NER model has no way to interact with the IDs used by the EntityRuler. You might want to look at the [EntityLinker](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson).
